Enterprise-CMCS/cmcs-eregulations
# EREGCSC-2041 -- Adjust SSA.gov Title 16 links

Resolves [EREGCSC-2041](https://jiraent.cms.gov/browse/EREGCSC-2041)

**Description**

1. Pulls titles from `/v3/acts` to be used for the StatuteSelector list items.

2. Fixes issue with Title 16 SSA.gov compilation page URLs.  

In the statute table, there is a column for links to the SSA.gov compilation page.  Normally, the links to this page only need the title number and the section.

Example: Title 11, Section 1101

✅ `https://www.ssa.gov/OP_Home/ssact/title11/1101.htm`

However, SSA.gov Title 16 links are a special case.  They need to have a `b` appended after the title number.  

Example: Title 16, Section 1601

❌ `https://www.ssa.gov/OP_Home/ssact/title16/1601.htm`

✅ `https://www.ssa.gov/OP_Home/ssact/title16b/1601.htm`

**This pull request changes:**

- Creates list of titles used in StatuteSelector component from `v3/acts` API endpoint response
- Adds condition in `ssaGovUrl` utility method that checks for the existence of Title 16 and, if found, uses `16b` as the title in the returned URL.
- Minor formatting and ESLint changes in `urlMethods.js`
- Moves some transformation/shaping logic from `Statutes.vue` to the shared `utils` file so those new methods can be used in component unit tests, as well as having unit tests of their own.

**Steps to manually verify this change:**

1. Visit the [experimental deployment ](https://4fe6sqd0ad.execute-api.us-east-1.amazonaws.com/dev910/statutes)
2. Make sure State Selector list (left side of page) loads fine and includes Title XVI (Title 16)
3. Click Title XVI
4. Make sure SSA.gov Compilation links contain `16b` instead of `16`
5. Visit these updated SSA.gov links and make sure they go where expected to go
6. Compare to [`prod` statute table with title 16 selected](https://regulations-pilot.cms.gov/statutes/?act=ssa&title=16)
